packages/volto/src/helpers/Robots/Robots.js

```diff
@@ -15,43 +17,23 @@ import { addHeadersFactory } from '@plone/volto/helpers/Proxy/Proxy';
  */
 export const generateRobots = (req) =>
   new Promise((resolve) => {
-    const internalUrl =
-      config.settings.internalApiPath ??
-      config.settings.devProxyToApiPath ??
-      config.settings.apiPath;
-    const request = superagent.get(`${internalUrl}/robots.txt`);
-    request.set('Accept', 'text/plain');
+    const request = superagent.get(formatUrl('@site'));
+    request.set('Accept', 'application/json');
     const authToken = req.universalCookies.get('auth_token');
     if (authToken) {
       request.set('Authorization', `Bearer ${authToken}`);
     }
     request.use(addHeadersFactory(req));
-    request.end((error, { text }) => {
+    request.end((error, { text, body }) => {
       if (error) {
         resolve(text || error);
       } else {
-        // It appears that express does not take the x-forwarded headers into
-        // consideration, so we do it ourselves.
-        const {
-          'x-forwarded-proto': forwardedProto,
-          'x-forwarded-host': forwardedHost,
-          'x-forwarded-port': forwardedPort,
-        } = req.headers;
-        const proto = forwardedProto ?? req.protocol;
-        const host = forwardedHost ?? req.get('Host');
-        const portNum = forwardedPort ?? req.get('Port');
-        const port =
-          (proto === 'https' && '' + portNum === '443') ||
-          (proto === 'http' && '' + portNum === '80')
-            ? ''
-            : `:${portNum}`;
-        // Plone has probably returned the sitemap link with the internal url.
-        // If so, let's replace it with the current one.
-        const url = `${proto}://${host}${port}`;
-        text = text.replace(internalUrl, url);
-        // Replace the sitemap with the sitemap index.
-        text = text.replace('sitemap.xml.gz', 'sitemap-index.xml');
```

Comment:
@sneridagh @ericof Removing this replacement is a breaking change. We should keep it here, or handle it instead in https://github.com/plone/plone.volto/pull/183

Reply:
Fixed, however I think it has to go to plone.volto too. Although that can be a breaking change, since it can mean an action needed in Google Search Consoles of people.

Reply:
@sneridagh No, it's not a breaking change. It was getting replaced before, and it still will be.